mesheryctl/internal/cli/root/system/start.go

```diff
@@ -98,155 +134,214 @@ func start() error {
 			} else {
-				// Assume Meshery running on MacOS host
-				err = exec.Command("open", url).Start()
-				if err != nil {
-					return errors.Wrap(err, utils.SystemError("failed to exec command"))
+				checkFlag = 1
+			}
+		}
+
+		//if meshery_meshery_1 failed to start showing logs
+		//code for logs
+		if checkFlag == 1 {
+			log.Info("Starting Meshery logging . . .")
+			cmdlog := exec.Command("docker-compose", "-f", utils.DockerComposeFile, "logs", "-f")
+			cmdReader, err := cmdlog.StdoutPipe()
+			if err != nil {
+				return errors.Wrap(err, utils.SystemError("failed to create stdout pipe"))
+			}
+			scanner := bufio.NewScanner(cmdReader)
+			go func() {
+				for scanner.Scan() {
+					log.Println(scanner.Text())
 				}
+			}()
+			if err := cmdlog.Start(); err != nil {
+				return errors.Wrap(err, utils.SystemError("failed to start logging"))
 			}
+			if err := cmdlog.Wait(); err != nil {
+				return errors.Wrap(err, utils.SystemError("failed to wait for command to execute"))
+			}
+		}
 
-			//check flag to check successful deployment
-			checkFlag = 0
-			break
-		} else {
-			checkFlag = 1
+	case "kubernetes":
+
+		config, err := meshkitkube.DetectKubeConfig()
+
+		if err != nil {
+			return errors.Wrap(err, "failed to detect Kube Config file")
 		}
-	}
 
-	//if meshery_meshery_1 failed to start showing logs
-	//code for logs
-	if checkFlag == 1 {
-		log.Info("Starting Meshery logging . . .")
-		cmdlog := exec.Command("docker-compose", "-f", utils.DockerComposeFile, "logs", "-f")
-		cmdReader, err := cmdlog.StdoutPipe()
+		clientSet, err := kubernetes.NewForConfig(config)
+
 		if err != nil {
-			return errors.Wrap(err, utils.SystemError("failed to create stdout pipe"))
+			return errors.Wrap(err, "error setting clientset")
 		}
-		scanner := bufio.NewScanner(cmdReader)
-		go func() {
-			for scanner.Scan() {
-				log.Println(scanner.Text())
+
+		client, err := meshkitkube.New(clientSet, *config)
+
+		if err != nil {
+			return errors.Wrap(err, "failed to create new client")
+		}
+
+		manifests, err := utils.ListManifests(manifestsURL)
+
+		if err != nil {
+			return errors.Wrap(err, "failed to make GET request")
+		}
+
+		if _, err := os.Stat(utils.ManifestsFolder); os.IsNotExist(err) {
+			if err := os.MkdirAll(filepath.Join(utils.MesheryFolder, utils.ManifestsFolder), os.ModePerm); err != nil {
+				return errors.Wrapf(err, utils.SystemError(fmt.Sprintf("failed to make %s directory", utils.ManifestsFolder)))
 			}
-		}()
-		if err := cmdlog.Start(); err != nil {
-			return errors.Wrap(err, utils.SystemError("failed to start logging"))
+			log.Info("created folder")
 		}
-		if err := cmdlog.Wait(); err != nil {
-			return errors.Wrap(err, utils.SystemError("failed to wait for command to execute"))
+
+		err = utils.DownloadManifests(manifests, rawManifestsURL)
+
+		if err != nil {
+			return errors.Wrap(err, "failed to download manifests")
 		}
+
+		err = ApplyManifestFiles(RequestedAdapters, client)
 	}
+
 	return nil
 }
```

Comment:
platform support for kubernetes. Please check the comments for details on specific functions.